packages/office-ui-fabric-react/src/components/DetailsList/DetailsList.base.tsx

```diff
@@ -415,8 +415,10 @@ const DetailsListInner: React.ComponentType<IDetailsListInnerProps> = (
       onRenderHeader: finalOnRenderDetailsGroupHeader,
       // pass through custom group header checkbox label
       headerProps: {
+        ...groupProps?.headerProps,
```

Comment:
We'll need to make the associated fix in `master` since it's slightly wrong there, too.